src/confluent_kafka/schema_registry/avro.py

```diff
@@ -228,27 +222,46 @@ def __init__(self, schema_registry_client, schema_str, to_dict=None, conf=None):
             raise ValueError("Unrecognized properties: {}"
                              .format(", ".join(conf_copy.keys())))
 
-        schema_dict = loads(schema.schema_str)
-        self._named_schemas = _resolve_named_schema(schema, schema_registry_client)
-        parsed_schema = parse_schema(schema_dict, named_schemas=self._named_schemas)
+        # Set common instance variables
+        # Called on __init__ and
+        # on __call__ if use.latest.version is set to True
+        self._set_instance_variables(schema_str)
+
+    def _set_instance_variables(self, schema_str, schema_id=None):
+        """
+        Function to be called upon __init__ and when serializing a message (__call__)
+        if use.latest.version is set True
```

Comment:
Add explanation of what this function does as well. This can be written after the description.

Reply:
Done